src/indent.ts

```diff
@@ -8,6 +8,38 @@ export function newlineAndIndent(
     edit: vscode.TextEditorEdit,
     args: any[]
 ) {
+    // Get rid of any user selected text, since a selection is
+    // always deleted whenever ENTER is pressed.
+    // This should always happen first
+    let selectedText = textEditor.document.getText(textEditor.selection);
+    if (selectedText) {
+        edit.delete(textEditor.selection);
+        // Make sure we get rid of the selection range.
+        // Find the earliest position of the selection
+        // and set selection start and end to it.
+        let lineNum = 0;
```

Comment:
I believe that the `textEditor.selection.start` is built to compute this location for you. Can we use that and get rid of most of this code?

Reply:
Wow much simpler! Tested and working as expected.